File: agents/javascript/src/tools/write-file.ts
/**
 * Tool: write_file
 * Writes content to a file, creating parent directories if needed.
 */

import * as fs from 'fs';
import * as path from 'path';

export async function writeFile(
  args: Record<string, unknown>,
  projectPath: string
): Promise<string> {
  const filePath = args.path as string;
  const content = args.content as string;

  const fullPath = path.resolve(projectPath, filePath);

  // Create parent directories
  fs.mkdirSync(path.dirname(fullPath), { recursive: true });

  // Write the file
  fs.writeFileSync(fullPath, content);

  const bytes = Buffer.byteLength(content);
  return `Written ${bytes} bytes to ${filePath}`;
}

File: agents/python/src/tools/write_file.py
"""Tool: write_file — write content to a file, creating dirs if needed."""

import os
from typing import Any


def write_file(args: dict[str, Any], project_path: str) -> str:
    file_path = args["path"]
    content = args["content"]

    full_path = os.path.normpath(os.path.join(project_path, file_path))
    os.makedirs(os.path.dirname(full_path), exist_ok=True)

    with open(full_path, "w", encoding="utf-8") as f:
        f.write(content)

    byte_count = len(content.encode("utf-8"))
    return f"Written {byte_count} bytes to {file_path}"
